backend/rankings_storage.py

```
@@ -0,0 +1,302 @@
+        {model_id: [{ts, rating, games}, ...]}. Empty dict if no log yet.
+    """
+    path = _matches_path(user_id)
+    if not path.exists():
+        return {}
+
+    ratings: dict[str, dict[str, Any]] = {}
+    history: dict[str, list[dict[str, Any]]] = {}
+
+    with open(path) as f:
+        for line in f:
+            line = line.strip()
```

Ruff (0.15.11)

[warning] `for` loop variable `line` overwritten by assignment target

(PLW2901)